api/src/main/java/org/apache/iceberg/actions/RewriteDataFiles.java

```diff
@@ -88,6 +89,21 @@ public interface RewriteDataFiles extends SnapshotUpdate<RewriteDataFiles, Rewri
   String USE_STARTING_SEQUENCE_NUMBER = "use-starting-sequence-number";
   boolean USE_STARTING_SEQUENCE_NUMBER_DEFAULT = true;
 
+  /**
+   * Forces the rewrite job order based on the value.
+   * <p><ul>
+   * <li> If rewrite.job-order=bytes-asc, then rewrite the smallest job groups first.
+   * <li> If rewrite.job-order=bytes-desc, then rewrite the largest job groups first.
+   * <li> If rewrite.job-order=files-asc, then rewrite the job groups with the least files first.
+   * <li> If rewrite.job-order=files-desc, then rewrite the job groups with the most files first.
+   * <li> If rewrite.job-order=none, then rewrite job groups in the order they were planned (no
+   * specific ordering).
+   * </ul><p>
+   * Defaults to none.
+   */
+  String REWRITE_JOB_ORDER = "rewrite.job-order";
```

Comment:
Why add a prefix and not just use `job-order`? Aren't all options rewrite specific?
cc @RussellSpitzer @rajarshisarkar

Reply:
we could do rewrite-job-order in the backport pr, I think that would be clearest if we want to avoid the sub prop

Reply:
Sure, I'll incorporate this change in the backport PR.